tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,201 @@
+    )
+    eso_version_after_upgrade = (
+        csv_after_upgrade.get('items')[1].get('spec').get('version')
+    )
+    if config.UPGRADE['upgrade_logging_channel'] in (
+        clo_version_after_upgrade and eso_version_after_upgrade
+    ):
+        logger.info(
+            f"Upgraded version of Cluster-logging "
+            f"operator {clo_version_after_upgrade}"
+        )
+        logger.info(
+            f"Upgraded version of Elastic-search "
+            f"operator {eso_version_after_upgrade}"
+        )
+        return True
+    return False
+
+
+@post_ocp_upgrade
+@pytest.mark.polarion_id("OCS-2201")
+class TestUpgradeLoggingAfterOCPUpgrade():
+    """
+    This class contains test for upgrade openshift-logging
+    after OCP upgrade
+    1. Monitors OCS health
+    2. Checks OCP version
+    3. Checks logging version, if mismatched then upgrades logging
+    4. Checks Elasticsearch and clusterlogging operator
+    5. Checks logging health post upgrade
+
+    """
+
+    def test_upgrade_logging(self):
+        """
+        This function contains test to upgrade openshift-logging
+        with Entry and Exit criteria for checks
+
+        """
+
+        ceph_cluster = CephCluster()
+        with CephHealthMonitor(ceph_cluster):
+
+            #  Pre-check
+            check_cluster_logging()
+
+            # Matching the OCP version and cluster-Logging version
+            ocp_version = get_ocp_version()
+            logger.info(f"OCP version {ocp_version}")
+            csv = ocp.OCP(
+                kind=constants.CLUSTER_SERVICE_VERSION,
+                namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+            )
+            csv_info = csv.get(
+                out_yaml_format=True
+            )
+            clo_version_before_upgrade = (
+                csv_info.get('items')[0].get('spec').get('version')
+            )
+            eso_version_before_upgrade = (
+                csv_info.get('items')[1].get('spec').get('version')
+            )
+            logger.info(
+                f"Cluster-logging operator version before upgrade "
+                f"{clo_version_before_upgrade}"
+            )
+            logger.info(
+                f"Elastic search operator version after upgrade "
+                f"{eso_version_before_upgrade}"
+            )
+            subscription = get_clusterlogging_subscription()
+            logging_channel = subscription.get('spec').get('channel')
+            logger.info(
+                f"Logging channel before upgrade {logging_channel}"
+            )
+            upgrade_channel = config.UPGRADE['upgrade_logging_channel']
+            if ocp_version > logging_channel:
+                # Upgrade Elastic search operator Subscription
+                es_subscription_cmd = (
+                    f'oc patch subscription elasticsearch-operator '
+                    f'-n {constants.OPENSHIFT_OPERATORS_REDHAT_NAMESPACE} '
+                    f'--type merge -p \'{{"spec":{{"channel": '
```

Comment:
here as well, so if you will remove `f` you will probably won't need to escape `{`